src/GitHubStatusDisplay.js

```diff
@@ -247,6 +259,22 @@ export default class BuildHistoryDisplay extends Component {
       <th className="rotate" key={jobName}><div className={consecutive_failure_count.has(jobName) ? "failing-header" : ""}>{summarize_job(jobName)}</div></th>
     );
 
+    const visible_job_groups = [];
+    let prevJob = null;
+    let jobGroup = [];
+    for (const foo of visible_jobs) {
+      if (prevJob && !sameGroup(prevJob, foo)) {
+        visible_job_groups.push(<colgroup className="job-group">{jobGroup}</colgroup>);
+        jobGroup = [];
+      }
+      prevJob = foo;
+      jobGroup.push(<col/>);
+    }
+    if (jobGroup) {
+      visible_job_groups.push(<colgroup className="job-group">{jobGroup}</colgroup>);
+      jobGroup = [];
+    }
+
```

Comment:
Does the above code assume a certain ordering of `visible_jobs`? I might be misunderstanding the code, but is it correct to say the jobs are already grouped and you're adding space around the groups?

Reply:
That is correct, yeah; a followup could also add some more logic to group `visible_jobs` more, which for instance would allow us to do groupings within the set of CircleCI jobs. For just GHA though, the current ordering is already what we need.